### PR TITLE
Added auto text direction

### PR DESCRIPTION
**Pull request by Amir-P**
From https://github.com/memspace/zefyr/pull/592

### CODE REVIEW

**Review by amantoux — CHANGES_REQUESTED** [review]
@Amir-P I've left some questions and a comment

File: packages/parchment/pubspec.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ dependencies:
   collection: ^1.16.0
   quill_delta: ^3.0.0-nullsafety.2
   quiver: ^3.1.0
+  intl: ^0.17.0
```
[the remark this answers is not in the document]
Comment (Amir-P) [reply packages/parchment/pubspec.yaml:13]: packages from Flutter depend on `intl` so it's pretty reliable. For example [`flutter_localizations`](https://github.com/flutter/flutter/blob/master/packages/flutter_localizations/pubspec.yaml).

File: packages/parchment/lib/src/heuristics/insert_rules.dart [hunk 2]
```diff
@@ -595,3 +597,45 @@ class MarkdownBlockShortcutsInsertRule extends InsertRule {
...
+      attributes = {
+        ...ParchmentAttribute.rtl.unset.toJson(),
+        ...ParchmentAttribute.alignment.unset.toJson(),
+      };
+    }
+
+    return Delta()
+      ..retain(index)
+      ..insert(data)
+      ..retain(1, attributes);
```
[the remark this answers is not in the document]
Comment (Amir-P) [reply packages/parchment/lib/src/heuristics/insert_rules.dart:639]: Since direction a line-scoped attribute, we're adding it to the new line character.

**Review by amantoux — APPROVED** [review]
LGTM

File: packages/parchment/lib/src/heuristics/insert_rules.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amir-P) [reply packages/parchment/lib/src/heuristics/insert_rules.dart:641]: Can you provide steps to reproduce and logs? @cgestes

[the remark this answers is not in the document]
Comment (cgestes) [reply packages/parchment/lib/src/heuristics/insert_rules.dart:641]: sorry was using a weird version of quill_delta. took me a bit to figure it out :)

All good :)

**Pull request merged**